go/src/koding/klient/machine/mount/index/index.go

```diff
@@ -0,0 +1,328 @@
+package index
+
+import (
+	"crypto/sha1"
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+// Entry represents a single file registered to index.
+type Entry struct {
+	Name  string      `json:"name"`  // The relative name of the file.
+	CTime int64       `json:"ctime"` // Metadata change time since EPOCH.
+	MTime int64       `json:"mtime"` // File data change time since EPOCH.
+	Mode  os.FileMode `json:"mode"`  // File mode and permission bits.
+	Size  int64       `json:"size"`  // Size of the file.
+	SHA1  []byte      `json:"sha1"`  // SHA-1 hash of file content.
+}
+
+// NewEntryFile creates new Entry from a file stored under path argument.
+// Info is optional and, if given, should store LStat result on the given file.
+func NewEntryFile(root, path string, info os.FileInfo) (e *Entry, err error) {
```

Comment:
From what I understand:

- we store relative paths, relative to the mount dir, so it's possible to use the same index for two mounts with different mount directories; the mount dir here is root (can we add a comment for this argument?)
- we store paths converted to Unix paths, so it'd possible to use window index on linux

Is it correct?